projectq/types/_qubit.py

```diff
@@ -83,7 +86,8 @@ def __eq__(self, other):
         Args:
             other (BasicQubit): BasicQubit to which to compare this one
         """
-        return (isinstance(other, self.__class__) and self.id == other.id and
```

Comment:
Using `isinstance(other, self.__class__)` means that `a == b` can return a different result than `b == a` when `b` is a subclass of `a`. Equality must *must* be commutative, so I changed `self.__class__` to `BasicQubit`.